tests/test_main.py

```diff
@@ -1,47 +1,179 @@
+def session_fixture():
+    """
+    Create an in-memory test database and return a session.
+    """
+    engine = create_engine(
+        "sqlite://",
+        connect_args={"check_same_thread": False},
+        poolclass=StaticPool,
+    )
+    UKRDC3Base.metadata.tables["patient"].create(bind=engine)
+    UKRDC3Base.metadata.tables["patientnumber"].create(bind=engine)
+    UKRDC3Base.metadata.tables["programmembership"].create(bind=engine)
+
+    with Session(engine) as session:
+        _create_test_data(session)
+        yield session
+
+
+@pytest.fixture(name="client")
+def client_fixture(session: Session):
+    """
+    Create a test client and override the FastAPI dependency.
+    Depends on the session fixture.
+    """
+    settings.apikeys = ["PYTESTKEY0000000000"]
 
-load_dotenv(find_dotenv())
+    def get_session_override():
+        return session
 
-client = TestClient(app)
+    app.dependency_overrides[get_session] = get_session_override
 
-fake_key = os.getenv("FAKEKEY")
-fake_identifier = os.getenv("FAKEIDENTIFIER")
+    client = TestClient(app)
+    yield client
+    app.dependency_overrides.clear()
 
 
-def test_main():
-    responce = client.get(
-        f"/radar_check/{fake_identifier}",
-        headers={
-            "Authorization": f"Bearer {fake_key}",
+def test_main_radar_check(client: TestClient):
+    response = client.post(
+        "/radar_check/",
+        headers={"Authorization": "Bearer PYTESTKEY0000000000"},
+        json={
+            "nhsNumber": "8888888888",
+            "dateOfBirth": "2000-01-01",
         },
     )
-    assert responce.status_code == 200
-    assert responce.json() == True
+    assert response.status_code == 200
+    assert response.json() == {"nhsNumber": True, "dateOfBirth": True}
 
 
-def test_main_no_patient():
-    responce = client.get(
-        "/radar_check/6",
-        headers={
-            "Authorization": f"Bearer {fake_key}",
+def test_main_radar_check_no_patient(client: TestClient):
+    response = client.post(
+        "/radar_check/",
+        headers={"Authorization": "Bearer PYTESTKEY0000000000"},
+        json={
+            "nhsNumber": "6",
+            "dateOfBirth": "1900-01-01",
         },
     )
-    assert responce.status_code == 200
-    assert responce.json() == False
+    assert response.status_code == 200
+    assert response.json() == {"nhsNumber": False, "dateOfBirth": False}
 
 
-def test_main_no_token():
-    response = client.get("/radar_check/1")
+def test_main_radar_check_nhs_correct_dob_not(client: TestClient):
+    response = client.post(
+        "/radar_check/",
+        headers={"Authorization": "Bearer PYTESTKEY0000000000"},
+        json={
+            "nhsNumber": "8888888888",
+            "dateOfBirth": "1900-01-01",
+        },
+    )
+    assert response.status_code == 200
+    assert response.json() == {"nhsNumber": True, "dateOfBirth": False}
+
+
+def test_main_radar_check_dob_correct_nhs_not(client: TestClient):
```

Comment:
Current code prevents this from being an issue but worth testing in case future devs change anything. This would be an IG concern.